fplll/nr/matrix.h

```diff
@@ -250,23 +250,26 @@ template <class ZT> class ZZ_mat : public Matrix<Z_NR<ZT>>
       initialized with the default constructor of Z_NR&lt;T&gt;. */
   ZZ_mat(int rows, int cols) : Matrix<T>(rows, cols) {}
 
-  // generators
+  /** Generate a zero matrix */
   void gen_zero(int d, int n)
   {
     resize(d, n);
     for (int i = 0; i < d; i++)
       matrix[i].fill(0);
   }
-
+  /** Generate an identity matrix */
   void gen_identity(int d)
   {
     gen_zero(d, d);
     for (int i     = 0; i < d; i++)
       matrix[i][i] = 1;
   }
 
+  /** Generate an augmented matrix of random coefficients for the first column */
   void gen_intrel(int bits);
+  /** Generate a matrix with coefficients of the first row and the diagonal */
```

Comment:
You can nick documentation for these from the command line `latticegen` tool and/or fpylll.